### PR TITLE
[]ASP.NET Core] Update `http.status_code` dimension type to int

### LINKED ISSUE #3777: Reconcile http.status_code type
As described in https://github.com/open-telemetry/opentelemetry-specification/issues/2855, the `http.status_code` type is inconsistent between Metrics and Traces. Opening this issue per discussion in the 2022-10-11 SIG as it is assumed that the type mismatch is unintentional and should be addressed pending the outcome of that issue.

### PR DESCRIPTION
**Pull request by vishweshbankwar**
Towards #3373 
Fixes #3777

## Changes
As per spec https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/semantic_conventions/http-metrics.md#attributes

For significant contributions please make sure you have completed the following items:

* [x] Appropriate `CHANGELOG.md` updated for non-trivial changes
* [ ] Design discussion issue #
* [ ] Changes in public API reviewed

### CODE REVIEW

**Review by CodeBlanch — APPROVED** [review]
LGTM

File: src/OpenTelemetry.Instrumentation.AspNetCore/Implementation/HttpInMetricsListener.cs [hunk 1]
```diff
@@ -63,7 +63,7 @@ public override void OnEventWritten(string name, object payload)
                 tags.Add(new KeyValuePair<string, object>(SemanticConventions.AttributeHttpFlavor, HttpTagHelper.GetFlavorTagValueFromProtocol(context.Request.Protocol)));
                 tags.Add(new KeyValuePair<string, object>(SemanticConventions.AttributeHttpScheme, context.Request.Scheme));
                 tags.Add(new KeyValuePair<string, object>(SemanticConventions.AttributeHttpMethod, context.Request.Method));
-                tags.Add(new KeyValuePair<string, object>(SemanticConventions.AttributeHttpStatusCode, context.Response.StatusCode.ToString()));
+                tags.Add(new KeyValuePair<string, object>(SemanticConventions.AttributeHttpStatusCode, context.Response.StatusCode));
```
Comment (utpilla) [inline src/OpenTelemetry.Instrumentation.AspNetCore/Implementation/HttpInMetricsListener.cs:66]: Use `(int)context.Response.StatusCode`? We have been explicitly type-casting it in other files.

Comment (vishweshbankwar) [reply src/OpenTelemetry.Instrumentation.AspNetCore/Implementation/HttpInMetricsListener.cs:66]: which file? we don't for ASP.NET Core as it is already `int`
https://github.com/open-telemetry/opentelemetry-dotnet/blob/c5ecd05c7220f012ca619305dd34ca0427605d4d/src/OpenTelemetry.Instrumentation.AspNetCore/Implementation/HttpInListener.cs#L247

Comment (utpilla) [reply src/OpenTelemetry.Instrumentation.AspNetCore/Implementation/HttpInMetricsListener.cs:66]: Ah I see. I got confused with the HttpInstrumentation files where the StatusCode is an enum. Never mind.

**Review by utpilla — APPROVED** [review]

**Review by Kielek — APPROVED** [review]

**Review by reyang — APPROVED** [review]

**Pull request merged**